docs/src/literate-tutorials/reactive_surface.jl

```diff
@@ -0,0 +1,259 @@
+    Me = zeros(2*n_basefuncs, 2*n_basefuncs)
+    De = zeros(2*n_basefuncs, 2*n_basefuncs)
+
+    ## Create an assembler
+    M_assembler = start_assemble(M)
+    D_assembler = start_assemble(D)
+    ## Loop over all cels
+    for cell in CellIterator(dh)
+        ## Reinitialize cellvalues for this cell
+        reinit!(cellvalues, cell)
+        ## Compute element contribution
+        assemble_element_mass!(Me, cellvalues)
+        assemble!(M_assembler, celldofs(cell), Me)
+
+        assemble_element_diffusion!(De, cellvalues)
+        assemble!(D_assembler, celldofs(cell), De)
+    end
+    return nothing
+end
+
+# ### Initial condition setup
+# Time-dependent problems always need an initial condition from which the time evolution starts.
+# In this tutorial we set the concentration of reactant 1 to $1$ and the concentration of reactant
+# 2 to $0$ for all nodal dof with associated coordinate $z \leq 0.9$ on the sphere. Since the
+# simulation would be pretty boring with a steady-state initial condition, we introduce some
+# heterogeneity by setting the dofs associated to top part of the sphere (i.e. dofs with $z > 0.9$
+# to store the reactant concentrations of $0.5$ and $0.25$ for the reactants 1 and 2 respectively.
+function setup_initial_conditions!(u₀::Vector, cellvalues::CellValues, dh::DofHandler)
+    u₀ .= ones(ndofs(dh))
+    u₀[2:2:end] .= 0.0
+
+    n_basefuncs = getnbasefunctions(cellvalues)
+
+    for cell in CellIterator(dh)
+        reinit!(cellvalues, cell)
+
+        coords = getcoordinates(cell)
+        dofs = celldofs(cell)
+        uₑ = @view u₀[dofs]
+        rv₀ₑ = reshape(uₑ, (2, n_basefuncs))
+
+        for i in 1:n_basefuncs
+            if coords[i][3] > 0.9
+                rv₀ₑ[1, i] = 0.5
+                rv₀ₑ[2, i] = 0.25
+            end
+        end
+    end
+
+    u₀ .+= 0.01*rand(ndofs(dh))
+end
+
+# ### Simulation routines
+# Now we define a function to setup and solve the problem with given feed and conversion rates
+# $F$ and $k$, as well as the time step length and for how long we want to solve the model.
+function gray_scott_sphere(F, k, Δt, T, refinements)
+    ## We start by setting up grid, dof handler and the matrices for the heat problem.
+    gmsh.initialize()
+
+    ## Add a unit sphere
+    gmsh.model.occ.addSphere(0.0,0.0,0.0,1.0)
+    gmsh.model.occ.synchronize()
+
+    gmsh.model.mesh.renumberNodes()
+    gmsh.model.mesh.renumberElements()
+
+    ## Generate surface elements and refine several times
+    gmsh.model.mesh.generate(2)
+    for _ in 1:refinements
+        gmsh.model.mesh.refine()
+    end
+
+    ## Create a grid out of it
+    nodes = tonodes()
+    elements, _ = toelements(2)
+    gmsh.finalize()
+    grid = Grid(elements, nodes);
+
+    ip = Lagrange{RefTriangle, 1}()
+    ip_geo = Lagrange{RefTriangle, 1}()
+    qr = QuadratureRule{RefTriangle}(2)
+    cellvalues = CellValues(qr, ip, ip_geo^3);
+
+    dh = DofHandler(grid);
+    add!(dh, :reactants, ip^2);
```

Comment:
Does it make sense to think of this as a vector field? Perhaps it should be two scalar fields?

Reply:
This is a technical trick to have a fast query for the nodal dofs. See the updated docs for why we need this.

Reply:
Otherwise it would just be the transpose though, but doesn't really matter.